connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/SparkSessionCleanRoomE2ESuite.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+import org.scalatest.concurrent.Eventually._
+
+import org.apache.spark.sql.connect.client.util.RemoteSparkSession
+import org.apache.spark.util.ThreadUtils
+
+/**
+ * Warning: SPARK-43648 moves two test cases related to `interrupt all` from `ClientE2ETestSuite`
+ * to the current test class to avoid the maven test issue of missing
+ * `org.apache.spark.sql.connect.client.SparkResult` during udf deserialization in server side. So
+ * please don't import classes that only exist in `spark-connect-client-jvm.jar` into the this
+ * class, as it will trigger similar maven test failures again.
+ */
```

Comment:
nit: for readers in the future, this exact history is irrelevant.
The comment should just say that this suite should not import any classes from `spark-connect-client-jvm.jar` because it could trigger serialization error.
It's enough to refer to SPARK-43648 for anyone wanting to read more context.

Reply:
ok